pkgs/development/web/cypress/default.nix

```diff
@@ -14,13 +14,20 @@
 , xorg
 }:
 
-stdenv.mkDerivation rec {
+let
+  platform = if (stdenv.targetPlatform.isAarch64 && stdenv.targetPlatform.isLinux)
+             then "linux-arm64"
+             else "linux-x64";
+  checksum = if (stdenv.targetPlatform.isAarch64 && stdenv.targetPlatform.isLinux)
+             then "sha256-gcf/MJ5aNUPoH6qz0n9vjviTec1rcxB0UzF+++6bUTs="
+             else "sha256-26mkizwkF0qPX2+0rkjep28ZuNlLGPljCvVO73t34Lk=";
```

Comment:
Nice addition. Just some suggestions to make this a bit more correct and nicer for users when their platform is unsupported:

Instead of `targetPlatform`, could you use `hostPlatform`? `target` would only be relevant for compilers.

Instead of `if then else`, could you use a lookup table with an error message? Something like:

```nix
let
  allBinaries = {
    x86_64-linux = {
      platform = "linux-x64";
      checksum = ...;
    };
  };
  inherit (stdenv.hostPlatform) system;
  binaries = allBinaries.${system} or (throw "cypress: No binaries available for system ${system}");
  inherit (binaries) platform checksum;
```

Reply:
Thanks for the review and the suggestions.

1. Good point about `targetPlatform` and `hostPlatform`. I fixed it.
2. Your approach with the lookup table is indeed more elegant. It comes at the cost of additional complexity, but feels like the more appropriate solution. I applied it with some slight naming tweaks.

As far as I can tell, the changes do not affect the resulting derivation so there was no need to rebuild anything on my test machine.

FYI: Changes were applied to the original commit with a force push.